unitylibs/core/workflow/workflow-acrobat/upload-handler.js

```diff
@@ -48,6 +48,25 @@ export default class UploadHandler {
     return blob;
   }
 
+  async uploadFileToUnityWithRetry(url, blobData, fileType, assetId) {
+    let retryDelay = 1000;
+    const maxRetries = 3;
+    let error = null;
+    for (let attempt = 1; attempt <= maxRetries; attempt++) {
+        try {
+            const response = await this.uploadFileToUnity(url, blobData, fileType, assetId);
+            if (response.ok) {  return response; }
+        } catch (err) { error = err;}
+        if (attempt < maxRetries) {
+            await new Promise(resolve => setTimeout(resolve, retryDelay));
+            retryDelay *= 2;
+        }
+    }
+    if (error) { error.message = error.message + ', Max retry delay exceeded during upload';
+    } else {  error = new Error('Max retry delay exceeded during upload');  }
+    throw error 
+  }
+
```

Comment:
We can avoid creating an extra variable error using below. Also removed curly braces where there was single statement inside if
```suggestion
  async uploadFileToUnityWithRetry(url, blobData, fileType, assetId) {
    const maxRetries = 3;
    let retryDelay = 1000;
    for (let attempt = 1; attempt <= maxRetries; attempt++) {
      try {
        const response = await this.uploadFileToUnity(url, blobData, fileType, assetId);
        if (response.ok) return response;
      } catch (err) {
        if (attempt === maxRetries) {
          err.message += ', Max retry delay exceeded during upload';
          throw err;
        }
      }
      if (attempt < maxRetries) {
        await new Promise(resolve => setTimeout(resolve, retryDelay));
        retryDelay *= 2;
      }
    }
    throw new Error('Max retry delay exceeded during upload');
}

```